td/views.py

```diff
@@ -31,25 +29,20 @@
 from .utils import DataTableSourceView, svg_to_pdf
 
 
+@cache_page(60 * 1440, key_prefix="langcodestext")
 def codes_text_export(request):
     return HttpResponse(Language.codes_text(), content_type="text/plain")
 
 
+@cache_page(60 * 1440, key_prefix="langnamestext")
 def names_text_export(request):
     return HttpResponse(Language.names_text(), content_type="text/plain")
 
 
+@cache_page(60 * 1440, key_prefix="langnamesdata")
 def names_json_export(request):
-    data = cache_get_or_set("langnames", Language.names_data)
-    return JsonResponse(data, safe=False)  # Set safe to False to allow list instead of dict to be returned
```

Comment:
We need to be sure we are not using this cached data by name anywhere else.  Also note, that the cached data here will be different than that is cached with `cache_page`.